apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/CodegenLayout.kt

```diff
@@ -17,13 +20,62 @@ import com.apollographql.apollo3.compiler.singularize
  *
  * Inputs should always be GraphQL identifiers and outputs are valid Kotlin/Java identifiers.
  */
-abstract class CodegenLayout(
+internal abstract class CodegenLayout(
+    private val ir: Ir,
     private val packageNameGenerator: PackageNameGenerator,
     private val schemaPackageName: String,
     private val useSemanticNaming: Boolean,
     private val useSchemaPackageNameForFragments: Boolean,
-    private val nameToClassName: (String) -> String,
 ) {
+  private val schemaTypeToClassName: Map<IrSchemaType, String> = computeClassNames()
```

Comment:
Do you need to key by `IrSchemaType` instead of just `name`? I'd prefer keying by `name` because it makes it explicit that only this is used. Also allows to mutate the `Ir` if we need to down the road while maintaining stable keys.

Reply:
Oh yeah, we can keep the name keys 👍